pkg/compact/compact.go

```diff
@@ -897,7 +897,48 @@ func NewBucketCompactor(logger log.Logger, sy *Syncer, comp tsdb.Compactor, comp
 }
 
 // Compact runs compaction over bucket.
-func (c *BucketCompactor) Compact(ctx context.Context) error {
+func (c *BucketCompactor) Compact(ctx context.Context, concurrency int) error {
+	// Set up workers who will compact the groups when the groups are ready.
+	var wg sync.WaitGroup
+	defer wg.Wait()
+
+	groupChan := make(chan *Group)
+	errChan := make(chan error, concurrency)
+	var finishedAllGroups bool
+	var mtx sync.Mutex
+
+	workCtx, cancel := context.WithCancel(ctx)
```

Comment:
I feel like using https://godoc.org/golang.org/x/sync/errgroup could simplify this and remove the need for errChannel?

Please consider using that if possible. If not please handle conxtexts correctly in new goroutine.

Reply:
:+1: